### ISSUE TITLE
Vulnerability alerts not found in GitHub Enterprise

### ISSUE THREAD

**Issue by eocantu** [renovatebot/renovate#15916]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.10.2

### Please select which platform you are using if self-hosting.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.2.10

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

When we upgraded Renovate to version 32.10.2 (from 25.26.3) it stopped finding vulnerability alerts.

Renovate logs seem to indicate it is attempting to retrieve them but nothing comes back:

<details><summary>Logs: Renovate not finding vulnerabilities</summary>

```
TRACE: Performing Github GraphQL request (repository=eocantu/cdn-publish)
TRACE: Applying Bearer authentication (repository=eocantu/cdn-publish)
       "url": "https://github.mycompany.com/api/graphql"
TRACE: got request (repository=eocantu/cdn-publish)
       "url": "https://github.mycompany.com/api/graphql",
       "options": {
         "method": "post",
         "context": {"hostType": "github"},
         "hostType": "github",
         "baseUrl": "https://github.mycompany.com/api/",
         "headers": {
           "accept": "application/vnd.github.vixen-preview+json, application/vnd.github.machine-man-preview+json",
           "user-agent": "WhiteSource Renovate",
           "authorization": "***********"
         },
         "json": {
           "query": "\nquery($owner: String!, $name: String!) {\n  repository(owner: $owner, name: $name) {\n    vulnerabilityAlerts(last: 100, states: [OPEN]) {\n      edges {\n        node {\n          dismissReason\n          vulnerableManifestFilename\n          vulnerableManifestPath\n          vulnerableRequirements\n          securityAdvisory {\n            description\n            identifiers { type value }\n            references { url }\n            severity\n          }\n          securityVulnerability {\n            package { name ecosystem }\n            firstPatchedVersion { identifier }\n            vulnerableVersionRange\n          }\n        }\n      }\n    }\n  }\n}\n",
           "variables": {"owner": "eocantu", "name": "cdn-publish"}
         },
         "responseType": "json",
         "throwHttpErrors": true,
         "hooks": {"beforeRedirect": ["[function]"]},
         "timeout": 60000
       }
DEBUG: No vulnerability alerts found (repository=eocantu/cdn-publish)
DEBUG: No vulnerability alerts found (repository=eocantu/cdn-publish)
```

</details>

In fact, there was a security vulnerability PR open prior to the Renovate upgrade and it got closed.

<details><summary>Logs: Renovate closing existing vulnerability PR</summary>

```
DEBUG: Found PR #3 (repository=eocantu/cdn-publish)
DEBUG: Branch has not been modified (repository=eocantu/cdn-publish)
       "branchName": "renovate-npm-minimist-vulnerability"
 INFO: Autoclosing PR (repository=eocantu/cdn-publish)
       "branchName": "renovate-npm-minimist-vulnerability",
       "prNo": 3,
       "prTitle": "Fix(deps): Update dependency minimist to v1.2.6 [SECURITY]"
DEBUG: updatePr(3, Fix(deps): Update dependency minimist to v1.2.6 [SECURITY] - autoclosed, body) (repository=eocantu/cdn-publish)
TRACE: Applying Bearer authentication (repository=eocantu/cdn-publish)
       "url": "https://github.mycompany.com/api/v3/repos/eocantu/cdn-publish/pulls/3"
TRACE: got request (repository=eocantu/cdn-publish)
       "url": "https://github.mycompany.com/api/v3/repos/eocantu/cdn-publish/pulls/3",
       "options": {
         "method": "patch",
         "context": {"hostType": "github"},
         "hostType": "github",
         "baseUrl": "https://github.mycompany.com/api/v3/",
         "json": {
           "title": "Fix(deps): Update dependency minimist to v1.2.6 [SECURITY] - autoclosed",
           "state": "closed"
         },
         "responseType": "json",
         "throwHttpErrors": true,
         "headers": {
           "accept": "application/vnd.github.machine-man-preview+json",
           "user-agent": "WhiteSource Renovate",
           "authorization": "***********"
         },
         "hooks": {"beforeRedirect": ["[function]"]},
         "timeout": 60000
       }
DEBUG: PR updated (repository=eocantu/cdn-publish)
       "pr": 3
DEBUG: Deleted remote branch (repository=eocantu/cdn-publish)
       "branchName": "renovate-npm-minimist-vulnerability"
DEBUG: No local branch to delete (repository=eocantu/cdn-publish)
       "branchName": "renovate-npm-minimist-vulnerability"
TRACE: ensureIssueClosing(Action Required: Fix Renovate Configuration) (repository=eocantu/cdn-publish)
```

</details>

When I use the GraphQL query shown in the Renovate logs against the GitHub Enterprise server, it fails saying `Field 'vulnerabilityAlerts' doesn't accept argument 'states'`:

<details><summary>Logs: GraphQL query failing when using states argument</summary>

```
$ gh api graphql -F owner='eocantu' -F name='cdn-publish' -f query='query($owner: String!, $name: String!) {
>     repository(owner: $owner, name: $name) {
>         vulnerabilityAlerts(last: 100, states: [OPEN]) {
>             edges {
>                 node {
>                     dismissReason
>                     vulnerableManifestFilename
>                     vulnerableManifestPath
>                     vulnerableRequirements
>                     securityAdvisory {
>                         description
>                         identifiers {
>                             type
>                             value
>                         }
>                         references {
>                             url
>                         }
>                         severity
>                     }
>                     securityVulnerability {
>                         package {
>                             name
>                             ecosystem
>                         }
>                         firstPatchedVersion {
>                             identifier
>                         }
>                         vulnerableVersionRange
>                     }
>                 }
>             }
>         }
>     }
> }'
{
  "errors": [
    {
      "path": [
        "query",
        "repository",
        "vulnerabilityAlerts",
        "states"
      ],
      "extensions": {
        "code": "argumentNotAccepted",
        "name": "vulnerabilityAlerts",
        "typeName": "Field",
        "argumentName": "states"
      },
      "locations": [
        {
          "line": 3,
          "column": 40
        }
      ],
      "message": "Field 'vulnerabilityAlerts' doesn't accept argument 'states'"
    }
  ]
}
gh: Field 'vulnerabilityAlerts' doesn't accept argument 'states'
```

</details>

If I removed the `states` argument from the GraphQL query, it works:

<details><summary>Logs: GraphQL query succeeding without states argument</summary>

```
$ gh api graphql -F owner='eocantu' -F name='cdn-publish' -f query='query($owner: String!, $name: String!) {
>     repository(owner: $owner, name: $name) {
>         vulnerabilityAlerts(last: 100) {
>             edges {
>                 node {
>                     dismissReason
>                     vulnerableManifestFilename
>                     vulnerableManifestPath
>                     vulnerableRequirements
>                     securityAdvisory {
>                         description
>                         identifiers {
>                             type
>                             value
>                         }
>                         references {
>                             url
>                         }
>                         severity
>                     }
>                     securityVulnerability {
>                         package {
>                             name
>                             ecosystem
>                         }
>                         firstPatchedVersion {
>                             identifier
>                         }
>                         vulnerableVersionRange
>                     }
>                 }
>             }
>         }
>     }
> }'
{
  "data": {
    "repository": {
      "vulnerabilityAlerts": {
        "edges": [
          {
            "node": {
              "dismissReason": null,
              "vulnerableManifestFilename": "package-lock.json",
              "vulnerableManifestPath": "package-lock.json",
              "vulnerableRequirements": "= 1.2.5",
              "securityAdvisory": {
                "description": "Minimist \u003c=1.2.5 is vulnerable to Prototype Pollution via file index.js, function setKey() (lines 69-95).",
                "identifiers": [
                  {
                    "type": "GHSA",
                    "value": "GHSA-xvch-5gv4-984h"
                  },
                  {
                    "type": "CVE",
                    "value": "CVE-2021-44906"
                  }
                ],
                "references": [
                  {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-44906"
                  },
                  {
                    "url": "https://github.com/advisories/GHSA-xvch-5gv4-984h"
                  }
                ],
                "severity": "CRITICAL"
              },
              "securityVulnerability": {
                "package": {
                  "name": "minimist",
                  "ecosystem": "NPM"
                },
                "firstPatchedVersion": {
                  "identifier": "1.2.6"
                },
                "vulnerableVersionRange": "\u003c 1.2.6"
              }
            }
          }
        ]
      }
    }
  }
}
```

</details>

Perhaps this was introduced by https://github.com/renovatebot/renovate/pull/14431 ([relevant code](https://github.com/renovatebot/renovate/blob/143d72c6004d1d9099c51f6d35899bd9f9d25289/lib/modules/platform/github/index.ts#L1560-L1565)). Our GitHub Enterprise version is `3.2.10`

```
DEBUG: Platform config (repository=eocantu/cdn-publish)
       "platformConfig": {
         "hostType": "github",
         "endpoint": "https://github.mycompany.com/api/v3/",
         "isGHApp": true,
         "isGhe": true,
         "gheVersion": "3.2.10"
       },
       "renovateUsername": "renovate-test[bot]"
```

Additionally, the API docs confirm that `states` query argument on `vulnerabilityAlerts` isn't available: https://docs.github.com/en/enterprise-server@3.2/graphql/reference/objects and I only see [it starting with GitHub Enterprise 3.5](https://docs.github.com/en/enterprise-server@3.5/graphql/reference/objects).

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Same as #15907 but we can use this from now on as it is a big report issue instead of discussion

**Comment by rarkins** [comment]
Closed by #15920
